Content.Server/Anomaly/Effects/ProjectileAnomalySystem.cs

```diff
@@ -38,29 +40,36 @@ private void OnSupercritical(EntityUid uid, ProjectileAnomalyComponent component
 
     private void ShootProjectilesAtEntities(EntityUid uid, ProjectileAnomalyComponent component, float severity)
     {
-        var xform = Transform(uid);
-        var projectilesShot = 0;
-        var range = component.ProjectileRange * severity;
-        var mobQuery = GetEntityQuery<MindComponent>();
-
-        foreach (var entity in _lookup.GetEntitiesInRange(uid, range, LookupFlags.Dynamic))
+        var projectileCount = (int) MathF.Round(MathHelper.Lerp(component.MinProjectiles, component.MaxProjectiles, severity));
+        var xformQuery = GetEntityQuery<TransformComponent>();
+        var mobQuery = GetEntityQuery<MobStateComponent>();
+        var xform = xformQuery.GetComponent(uid);
+
+        var inRange = _lookup.GetEntitiesInRange(uid, component.ProjectileRange * severity, LookupFlags.Dynamic).ToList();
+        _random.Shuffle(inRange);
+        var priority = new List<EntityUid>();
+        foreach (var entity in inRange)
         {
-            if (projectilesShot >= component.MaxProjectiles * severity)
-                return;
+            if (mobQuery.HasComponent(entity))
+                priority.Add(entity);
+        }
 
-            // Sentient entities are more likely to be shot at than non sentient
-            if (!mobQuery.HasComponent(entity) && !_random.Prob(component.TargetNonSentientChance))
-                continue;
+        Logger.Debug($"shots: {projectileCount}");
+        while (projectileCount > 0)
+        {
+            Logger.Debug($"{projectileCount}");
+            var target = priority.Any()
+                ? _random.PickAndTake(priority)
+                : _random.Pick(inRange);
```

Comment:
priority mobs can get double-targeted so not sure if that was your intention (they need removing from inRange when added to priority).

Reply:
i was fine with it double-targeting priority mobs.